### PR TITLE
FIX: twa

### PR DESCRIPTION
**Pull request by larsoner**
Fixes a little bug that can occur with `time_as_index` giving a result of `-1` with weird sample rates.

cc @mdclarke

### CODE REVIEW

**Comment by nordme** [comment]
Hit a `KeyError` during local SSS routine on this branch while preprocessing GenZ resting MEG dataset cc @ktavabi 


>   File "/home/nordme/GitHub/nordme_work_repo/genz_rs.py", line 91, in <module>
>     print_status=True,
> 
>   File "/home/nordme/GitHub/mnefun/mnefun/_mnefun.py", line 649, in do_processing
>     outs[ii] = func(p, subjects, run_indices)
> 
>   File "/home/nordme/GitHub/mnefun/mnefun/_mnefun.py", line 925, in run_sss
>     run_sss_locally(p, subjects, run_indices)
> 
>   File "/home/nordme/GitHub/mnefun/mnefun/_mnefun.py", line 1225, in run_sss_locally
>     raw_annot.orig_time = mne.annotations._handle_meas_date(raw.info['orig_time'])  # noqa
> 
> KeyError: 'orig_time'

**Comment by larsoner** [comment]
@nordme can you see if the latest commit fixes things?

**Comment by nordme** [comment]
@larsoner @ktavabi It works!

**Comment by ktavabi** [comment]
@nordme did the MNEFUN processing pipeline run all the way through?

File: mnefun/_mnefun.py [hunk 1]
```diff
@@ -1812,20 +1869,19 @@ def gen_inverses(p, subjects, run_indices):
...
-                    p.cov_rank != 'full':
-                kwargs = dict()
-                # The empty-room covariance can have a different spatial
-                # pattern
-                if 'rank' in get_args(regularize):
-                    kwargs['rank'] = 'full'
-                empty_cov = regularize(empty_cov, raw.info, **kwargs)
+            if p.force_erm_cov_rank_full and p.cov_method == 'empirical':
+                empty_cov = regularize(
+                    empty_cov, epochs.info, rank='full')
```
Comment (ktavabi) [inline mnefun/_mnefun.py]: Does it make sense to put in an assertion here to make sure covs have same rank?

File: mnefun/_mnefun.py [hunk 2]
```diff
@@ -1131,7 +1134,26 @@ def run_sss_positions(fname_in, fname_out, host='kasga', opts='-force',
     # concatenate hp pos file for split raw files if any
     data = []
     for f in fnames_out:
-        data.append(read_head_pos(op.join(pout, f)))
+        this_pos = read_head_pos(op.join(pout, f))
+        # sanitize the stupid thing; sometimes MaxFilter produces a line of all
```
[the remark this answers is not in the document]
Comment (larsoner) [reply mnefun/_mnefun.py]: MaxFilter bugs make me :(

File: mnefun/_mnefun.py [hunk 3]
```diff
@@ -1645,28 +1646,41 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
...
+            print('    Epoching data (decim=%s -> sfreq=%0.1f Hz).'
                   % (decim[si], new_sfreq))
         if new_sfreq not in sfreqs:
             if len(sfreqs) > 0:
                 warnings.warn('resulting new sampling frequency %s not equal '
                               'to previous values %s' % (new_sfreq, sfreqs))
             sfreqs.add(new_sfreq)
         if p.autoreject_thresholds:
+            assert len(p.autoreject_types) > 0
+            assert all(a in ('mag', 'grad', 'eeg', 'ecg', 'eog')
```
[the remark this answers is not in the document]
Comment (larsoner) [reply mnefun/_mnefun.py]: otherwise you `dict` has something in it it shouldn't, possibly a typo in a `key`

File: mnefun/_mnefun.py (hunk 1 above)
Comment (larsoner) [reply mnefun/_mnefun.py]: Actually this step forces it to be full rank, in case you want it to be that way

**Comment by larsoner** [comment]
Thanks for the look @ktavabi

**Pull request merged**